cmd/helm/repo_add.go

```diff
@@ -165,15 +166,17 @@ func (o *repoAddOptions) run(out io.Writer) error {
 	}
 
 	c := repo.Entry{
-		Name:                  o.name,
-		URL:                   o.url,
-		Username:              o.username,
-		Password:              o.password,
-		PassCredentialsAll:    o.passCredentialsAll,
-		CertFile:              o.certFile,
-		KeyFile:               o.keyFile,
-		CAFile:                o.caFile,
-		InsecureSkipTLSverify: o.insecureSkipTLSverify,
+		Name:                        o.name,
+		URL:                         o.url,
+		Username:                    o.username,
+		Password:                    o.password,
+		PassCredentialsAll:          o.passCredentialsAll,
+		CertFile:                    o.certFile,
+		KeyFile:                     o.keyFile,
+		CAFile:                      o.caFile,
+		InsecureSkipTLSverify:       o.insecureSkipTLSverify,
+		RepoProxyUrl:                o.repoProxyUrl,
+		DoNotUseEnvSettingsProxyUrl: settings.ProxyUrl,
```

Comment:
The DoNotUseEnvSettingsProxyUrl cannot be private because it needs to be used here

, and this package is main, which is not in the same package as repo.Entry.